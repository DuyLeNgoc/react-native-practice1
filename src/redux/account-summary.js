import {
	showLoading,
	hideLoading
} from 'redux/sharedData';

import UserService from 'network/UserService';

//=============================//
//      Action Types
//=============================//
export const SUCCESS = 'AccountSummarySuccess';
export const FAILED  = 'AccountSummaryFailed';

//=============================//
//      Action Creators
//=============================//
export function accountSummaryRequestSuccess(json) {
	return {
		type: SUCCESS,
		accountList: json
	};
}

export function accountSummaryRequestFailed(error) {
	return {
		type: FAILED,
		error: error.message
	};
}

//=============================//
//      Reducer
//=============================//
export const INITIAL_STATE = {
	error: '',
	isSuccess: false,
	accountList: []
};

export function accountListReducer(state = INITIAL_STATE, action) {
  switch (action.type) {
		case SUCCESS:
			console.log('### accountListReducer SUCCESS ');
			return {
				...state,
				error: '',
				isSuccess: true,
				accountList: action.accountList
			};
		case FAILED:
		console.log('### accountListReducer FAILED ');
			return {
				...state,
				error: action.error,
				isSuccess: false,
				accountList: []
			};
		default:
			return state;
  }
};

//=============================//
//4: Load Data
//=============================//
export function getList(userCredentials) {
  return (dispatch, getState) => {
		dispatch(showLoading());
    return UserService.getList(userCredentials)
    .then(json => {
				dispatch(hideLoading());
        dispatch(accountSummaryRequestSuccess(json));
    })
    .catch(error => {
			dispatch(hideLoading());
      dispatch(accountSummaryRequestFailed(error))
    });
  };
}
