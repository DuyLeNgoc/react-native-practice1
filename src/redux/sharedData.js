import { combineReducers } from 'redux';

//=============================//
//      Action Types
//=============================//
export const SHOW_LOADING = 'SHOW_LOADING';
export const HIDE_LOADING = 'HIDE_LOADING';

export const SHOW_ERROR   = 'SHOW_ERROR';
export const HIDE_ERROR   = 'HIDE_ERROR';

export const SAVE_USER_SESSION  = 'SAVE_USER_SESSION';
export const CLEAR_USER_SESSION = 'CLEAR_USER_SESSION';

//=============================//
//      Action Creators
//=============================//
export function showLoading() {
	return {
		type: SHOW_LOADING,
		loading: true
	};
}

export function hideLoading(json) {
	return {
		type: HIDE_LOADING,
		loading: false
	};
}

export function showError(error) {
	return {
		type: SHOW_ERROR,
		error: error.message
	};
}

export function hideError() {
	return {
		type: HIDE_ERROR,
		error: ''
	};
}

export function saveUser(user) {
	return {
		type: SAVE_USER_SESSION,
		user: user
	};
}

export function clearUser() {
	return {
		type: CLEAR_USER_SESSION,
		user: null
	};
}

//=============================//
//      Reducer
//=============================//
//reducer for loading
export const initialSateLoading = {
	loading: false,
	message: ''
};

export function loadingInfo(state = initialSateLoading, action) {
  switch (action.type) {
    case SHOW_LOADING:
			return {
				...state,
				loading: true,
				message: action.message
			};
		case HIDE_LOADING:
			return {
				...state,
				loading: false,
				message: ''
			};
	}
	return state;
};

//reducer for error
export function errorMessage(state = null, action) {
  switch (action.type) {
    case SHOW_ERROR:
			return action.message;
		case HIDE_ERROR:
			return null;
	}
	return state;
};

//reducer for user session
export function userSession(state = null, action) {
  switch (action.type) {
    case SAVE_USER_SESSION:
			return action.user;
		case CLEAR_USER_SESSION:
			return null;
	}
	return state;
};

//combineReducers
const reducers = {
  errorMessage,
  userSession,
  loadingInfo
}
export default combineReducers(reducers);
