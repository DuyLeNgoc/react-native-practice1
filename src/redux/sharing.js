//=============================//
//      Action Types
//=============================//
export const SHOW_LOADING = 'SHOW_LOADING';
export const HIDE_LOADING = 'HIDE_LOADING';
export const SHOW_ERROR   = 'SHOW_ERROR';
export const HIDE_ERROR   = 'HIDE_ERROR';
export const RESET_INPUT  = 'RESET_INPUT';

//=============================//
//      Action Creators
//=============================//
export function showLoading() {
	return {
		type: SHOW_LOADING,
		loading: true
	};
}

export function hideLoading(json) {
	return {
		type: HIDE_LOADING,
		loading: false
	};
}

export function showError(error) {
	return {
		type: SHOW_ERROR,
		error: error.message
	};
}

export function hideError() {
	return {
		type: HIDE_ERROR,
		error: ''
	};
}

export function resetInput() {
	return {
		type: RESET_INPUT
	};
}

//=============================//
//      Reducer
//=============================//
export const INITIAL_STATE = {
	error: '',
	loading: false
};

export default function sharedData(state = INITIAL_STATE, action) {
  switch (action.type) {
    case SHOW_LOADING:
			return {
				...state,
				loading: true,
				error: ''
			};
		case HIDE_LOADING:
			return {
				...state,
				loading: false,
				error: ''
			};
		case SHOW_ERROR:
			return {
				...state,
				loading: false,
				error: action.error
			};
		case HIDE_ERROR:
			console.log(action.error);
			return {
				...state,
				loading: false,
				error: ''
			};
		case RESET_INPUT:
			return {
				...state,
				loading: false,
				error: '',
				resetInput: true
			};
		default:
			return state;
  }
};
