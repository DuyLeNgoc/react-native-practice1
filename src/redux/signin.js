import AuthenticationService from 'network/AuthenticationService';

//=============================//
//      Action Types
//=============================//
const REQUEST = 'Request';
const SUCCESS = 'Success';
const FAILED  = 'Failed';

//=============================//
//      Action Creators
//=============================//
function loginRequest() {
	return {
		type: REQUEST
	};
}

function loginRequestSuccess(json) {
	return {
		type: SUCCESS,
		payload: json
	};
}

function loginRequestFailed(error) {
	return {
		type: FAILED,
		error: error.message
	};
}

//=============================//
//      Reducer
//=============================//
const INITIAL_STATE = {
	error: '',
	loading: false
};

export function signInReducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case REQUEST:
			return {
				...state,
				loading: true,
				error: ''
			};
		case SUCCESS:
			return {
				...state,
				loading: false,
				error: '',
				user: action.payload
			};
		case FAILED:
			console.log(action.error);
			return {
				...state,
				loading: false,
				error: action.error
			};
		default:
      return state;
  }
};

//=============================//
//4: Load Data
//=============================//
const userInfo = {
	trainee_id: 1,
	full_name: 'test',
	email: 'test@gmail.com',
	birthday: '1986-01-20',
	token: 'abcxyzwendsjkfjdsklfjkds'
};

export function login(userCredentials) {
  return (dispatch, getState) => {
    dispatch(loginRequest());
    return AuthenticationService.signin(userCredentials)
    .then(response => {
      return response.json();
    })
    .then(jsonTask => {
        dispatch(loginRequestSuccess(jsonTask));
    })
    .catch(error => {
      console.log('There has been a problem with your fetch operation: ' + error.message);
      dispatch(loginRequestFailed(error))
    });
  };
}
