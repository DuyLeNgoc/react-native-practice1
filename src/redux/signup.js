import {
	showLoading,
	hideLoading
} from 'redux/sharedData';

import AuthenticationService from 'network/AuthenticationService';

//=============================//
//      Action Types
//=============================//
export const SUCCESS = 'Success';
export const FAILED  = 'Failed';

//=============================//
//      Action Creators
//=============================//
export function signUpRequestSuccess() {
	return {
		type: SUCCESS
	};
}

export function signUpRequestFailed(error) {
	return {
		type: FAILED,
		error: error.message
	};
}

//=============================//
//      Reducer
//=============================//
export const INITIAL_STATE = {
	error: '',
	isSuccess: false
};

export function signUpReducer(state = INITIAL_STATE, action) {
  switch (action.type) {
		case SUCCESS:
			return {
				...state,
				error: '',
				isSuccess: true
			};
		case FAILED:
			return {
				...state,
				error: action.error,
				isSuccess: false
			};
		default:
			return state;
  }
};

//=============================//
//4: Load Data
//=============================//
export function signUp(userCredentials) {
  return (dispatch, getState) => {
		dispatch(showLoading());
    return AuthenticationService.signup(userCredentials)
    .then(json => {
				dispatch(hideLoading());
        dispatch(signUpRequestSuccess());
    })
    .catch(error => {
			dispatch(hideLoading());
      dispatch(signUpRequestFailed(error))
    });
  };
}
