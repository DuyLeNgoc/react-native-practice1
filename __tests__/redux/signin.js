import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

//Testing: integrate with enzyme
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {
	loginRequest,
	loginRequestSuccess,
	loginRequestFailed,
	REQUEST,
	SUCCESS,
	FAILED
} from 'redux/signin';
import { INITIAL_STATE as initialState } from 'redux/signin';

const MOCK_USERINFO = {
	trainee_id: 1,
	full_name: 'test',
	email: 'test@gmail.com',
	birthday: '1986-01-20',
	token: 'abcxyzwendsjkfjdsklfjkds'
};

const ERROR = {message: 'testing failed'}

describe('SignIn Action Creators ', () => {
	it('creates a SUCCESS action', () => {
	  // You do it
	  expect(loginRequestSuccess(MOCK_USERINFO)).toEqual(
	    {
				type: SUCCESS,
				payload: MOCK_USERINFO
	    }
	  );
	  // Jest does it
	  expect(loginRequestSuccess(MOCK_USERINFO)).toMatchSnapshot();
	});

	it('creates a REQUEST action', () => {
	  // You do it
	  expect(loginRequest()).toEqual(
	    {
				type: REQUEST
	    }
	  );
	  // Jest does it
	  expect(loginRequest()).toMatchSnapshot();
	});

	it('creates a REQUEST action', () => {
	  // You do it
	  expect(loginRequestFailed(ERROR)).toEqual(
	    {
				type: FAILED,
				error: ERROR.message
	    }
	  );
	  // Jest does it
	  expect(loginRequestFailed(ERROR)).toMatchSnapshot();
	});
});
