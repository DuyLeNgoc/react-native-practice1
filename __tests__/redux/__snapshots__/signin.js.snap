// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Action Creators  creates a FAILED action 1`] = `
Object {
  "error": "testing failed",
  "type": "SignInFailed",
}
`;

exports[`SignIn Action Creators  creates a SUCCESS action 1`] = `
Object {
  "payload": Object {
    "birthday": "1986-01-20",
    "email": "test@gmail.com",
    "full_name": "test",
    "token": "abcxyzwendsjkfjdsklfjkds",
    "trainee_id": 1,
  },
  "type": "SignInSuccess",
}
`;
